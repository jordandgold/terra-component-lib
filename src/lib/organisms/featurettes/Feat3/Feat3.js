import React, { Component } from "react";
import PropTypes from "prop-types";
import IconListItem from "../../../IconListItem/IconListItem";
import LinkCard from "../../../LinkCard/LinkCard";
import ButtonLink from "../../../ButtonLink/ButtonLink";
import "./Feat3.scss";

class Feat3 extends Component {
  getContent = () => {
    if (this.props.content[0].link) {
      return this.getLinkCards();
    } else {
      return this.getIconListItems();
    }
  };

  getLinkCards = () => {
    return this.props.content.map((item, index) => {
      return (
        <LinkCard
          title={item.title}
          text={item.text}
          link={item.link}
          key={`link-card-${index}`}
        />
      );
    });
  };

  getIconListItems = () => {
    return this.props.content.map((item, index) => {
      return <IconListItem content={item} key={`icon-list-item-${index}`} />;
    });
  };

  render() {
    const { images, text, title, cta } = this.props;
    return (
      <section className="ter-feat-three">
        <div className="ter-feat-three__image-wrapper">
          <img
            src={images.desktop.url}
            alt={images.desktop.altText}
            className="ter-feat-three__image ter-feat-three__image--desktop"
          />
          <img
            src={images.tablet.url}
            alt={images.tablet.altText}
            className="ter-feat-three__image ter-feat-three__image--tablet"
          />
          <img
            src={images.mobile.url}
            alt={images.mobile.altText}
            className="ter-feat-three__image ter-feat-three__image--mobile"
          />
        </div>
        <div className="ter-feat-three__content-container">
          <h2 className="ter-feat-three__title">{title}</h2>
          <p className="ter-feat-three__text">{text}</p>
          {this.getContent()}
          {cta && (
            <ButtonLink
              link={cta.link}
              text={cta.text}
              className="ter-button--primary--1"
            />
          )}
        </div>
      </section>
    );
  }
}

export default Feat3;

Feat3.propTypes = {
  images: PropTypes.shape({
    desktop: PropTypes.shape({
      url: PropTypes.string.isRequired,
      altText: PropTypes.string.isRequired
    }),
    mobile: PropTypes.shape({
      url: PropTypes.string.isRequired,
      altText: PropTypes.string.isRequired
    }),
    tablet: PropTypes.shape({
      url: PropTypes.string.isRequired,
      altText: PropTypes.string.isRequired
    }),
    content: PropTypes.arrayOf(
      PropTypes.shape({
        title: PropTypes.string.isRequired,
        text: PropTypes.string.isRequired,
        link: PropTypes.shape({
          url: PropTypes.string.isRequired,
          text: PropTypes.string.isRequired
        })
      })
    ),
    cta: PropTypes.shape({
      link: PropTypes.string.isRequired,
      text: PropTypes.string.isRequired
    })
  })
};
