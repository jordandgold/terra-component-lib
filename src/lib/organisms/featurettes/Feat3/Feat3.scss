@import "../../../global-styles/terra.scss";

.ter-feat-three {
  padding: spacing("lg") spacing("md");

  @media (min-width: $breakpoint-md) {
    @include grid-container;
    align-items: center;
  }

  &__image--desktop {
    display: none;

    @media (min-width: $breakpoint-md) {
      display: block;
    }
  }

  &__image--tablet {
    display: none;

    @media (min-width: $breakpoint-xs) {
      display: block;
    }

    @media (min-width: $breakpoint-md) {
      display: none;
    }
  }

  &__image--mobile {
    @media (min-width: $breakpoint-xs) {
      display: none;
    }
  }

  &__image-wrapper {
    margin-bottom: 0;

    @media (min-width: $breakpoint-md) {
      @include grid-column(6);
    }
  }

  &__content-container {
    @media (min-width: $breakpoint-md) {
      @include grid-column(6);
    }
  }

  &__image {
    width: 100%;
  }

  &__title {
    font-size: font-size("lg");
    margin-top: spacing("md");
  }

  .ter-icon-list-item {
    display: block;

    @media (min-width: $breakpoint-xs) {
      display: flex;
    }
  }

  &__text {
    @media (min-width: $breakpoint-md) {
      margin-bottom: 0;
    }
  }

  .ter-icon-list-item {
    margin-top: spacing("lg");
    padding: 0;

    &__title {
      margin-top: spacing("mini");
      margin-bottom: 0;
      font-size: font-size("md");
    }

    &__body {
      margin-top: spacing("xxs");

      @media (min-width: $breakpoint-md) {
        margin-bottom: 0;
      }
    }
  }

  .ter-link-card {
    margin-top: spacing("md");

    &__text {
      font-size: font-size("base");
    }

    &__title {
      font-size: font-size("md");
    }

    &__link {
      font-size: font-size("base");
    }

    .ter-icon {
      transform: translateY(9px) scale(0.7);
    }
  }

  .ter-button {
    display: block;

    @media (min-width: $breakpoint-xs) {
      display: inline-block;
      padding-left: spacing("lg");
      padding-right: spacing("lg");
    }

    margin-top: spacing("lg");
  }
}
