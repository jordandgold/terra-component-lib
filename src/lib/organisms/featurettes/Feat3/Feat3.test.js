import React from "react";
import { shallow } from "enzyme";
import Feat3 from "./Feat3";
import {
  mockFeat3Content,
  mockFeat3ContentTwo
} from "../../../../stories/mockContent";

describe("Feat3", () => {
  let wrapper;

  beforeEach(() => {
    const { title, text, content, cta, images } = mockFeat3ContentTwo;

    wrapper = shallow(
      <Feat3
        title={title}
        text={text}
        content={content}
        cta={cta}
        images={images}
      />
    );
  });

  it("should match the snapshot with links", () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe("getContent", () => {
    it("should call getLinkCards", () => {
      const spy = jest.spyOn(wrapper.instance(), "getLinkCards");

      wrapper.instance().getLinkCards();

      expect(spy).toHaveBeenCalled();
    });
  });

  describe("getLinkCards", () => {
    it("should return some jsx", () => {
      const expected = [
        '<article class="ter-link-card"><h4 class="ter-link-card__title">Good design is innovative</h4><p class="ter-link-card__text">Cupidatat ea in duis ullamco nostrud cupidatat quis aliqua ex labore occaecat sit. Tempor irure non amet fugiat qui ea occaecat qui. Aliquip esse eu exercitation veniam irure consectetur quis cupidatat incididunt reprehenderit sint.</p><a class="ter-link-card__link" url="#">Button<svg viewBox="0 0 8 8" class="ter-icon ter-icon--enclosed-arrow-right-dark-32px ter-icon--32px undefined"><use xlink:href="[object Object]#ter-icon--enclosed-arrow-right-dark-32px"></use></svg></a></article>',
        '<article class="ter-link-card"><h4 class="ter-link-card__title">Good design is innovative</h4><p class="ter-link-card__text">Cupidatat ea in duis ullamco nostrud cupidatat quis aliqua ex labore occaecat sit. Tempor irure non amet fugiat qui ea occaecat qui. Aliquip esse eu exercitation veniam irure consectetur quis cupidatat incididunt reprehenderit sint.</p><a class="ter-link-card__link" url="#">Button<svg viewBox="0 0 8 8" class="ter-icon ter-icon--enclosed-arrow-right-dark-32px ter-icon--32px undefined"><use xlink:href="[object Object]#ter-icon--enclosed-arrow-right-dark-32px"></use></svg></a></article>',
        '<article class="ter-link-card"><h4 class="ter-link-card__title">Good design is innovative</h4><p class="ter-link-card__text">Cupidatat ea in duis ullamco nostrud cupidatat quis aliqua ex labore occaecat sit. Tempor irure non amet fugiat qui ea occaecat qui. Aliquip esse eu exercitation veniam irure consectetur quis cupidatat incididunt reprehenderit sint.</p><a class="ter-link-card__link" url="#">Button<svg viewBox="0 0 8 8" class="ter-icon ter-icon--enclosed-arrow-right-dark-32px ter-icon--32px undefined"><use xlink:href="[object Object]#ter-icon--enclosed-arrow-right-dark-32px"></use></svg></a></article>'
      ];
      const response = wrapper
        .instance()
        .getLinkCards()
        .map(item => {
          return shallow(item).html();
        });

      expect(response).toEqual(expected);
    });
  });

  describe("getIconListItems", () => {
    it("should return some jsx", () => {
      const expected = [
        '<article class="ter-icon-list-item"><div class="ter-icon-list-item__icon-wrapper"><svg viewBox="0 0 8 8" class="ter-icon ter-icon--enclosed-arrow-right-dark-32px ter-icon--32px undefined"><use xlink:href="[object Object]#ter-icon--enclosed-arrow-right-dark-32px"></use></svg></div><div class="ter-icon-list-item__content-wrapper"><h4 class="ter-icon-list-item__title">Good design is innovative</h4><p class="ter-icon-list-item__body">Cupidatat ea in duis ullamco nostrud cupidatat quis aliqua ex labore occaecat sit. Tempor irure non amet fugiat qui ea occaecat qui. Aliquip esse eu exercitation veniam irure consectetur quis cupidatat incididunt reprehenderit sint.</p></div></article>',
        '<article class="ter-icon-list-item"><div class="ter-icon-list-item__icon-wrapper"><svg viewBox="0 0 8 8" class="ter-icon ter-icon--enclosed-arrow-right-dark-32px ter-icon--32px undefined"><use xlink:href="[object Object]#ter-icon--enclosed-arrow-right-dark-32px"></use></svg></div><div class="ter-icon-list-item__content-wrapper"><h4 class="ter-icon-list-item__title">Good design is innovative</h4><p class="ter-icon-list-item__body">Cupidatat ea in duis ullamco nostrud cupidatat quis aliqua ex labore occaecat sit. Tempor irure non amet fugiat qui ea occaecat qui. Aliquip esse eu exercitation veniam irure consectetur quis cupidatat incididunt reprehenderit sint.</p></div></article>',
        '<article class="ter-icon-list-item"><div class="ter-icon-list-item__icon-wrapper"><svg viewBox="0 0 8 8" class="ter-icon ter-icon--enclosed-arrow-right-dark-32px ter-icon--32px undefined"><use xlink:href="[object Object]#ter-icon--enclosed-arrow-right-dark-32px"></use></svg></div><div class="ter-icon-list-item__content-wrapper"><h4 class="ter-icon-list-item__title">Good design is innovative</h4><p class="ter-icon-list-item__body">Cupidatat ea in duis ullamco nostrud cupidatat quis aliqua ex labore occaecat sit. Tempor irure non amet fugiat qui ea occaecat qui. Aliquip esse eu exercitation veniam irure consectetur quis cupidatat incididunt reprehenderit sint.</p></div></article>'
      ];
      const { title, text, content, cta, images } = mockFeat3ContentTwo;

      wrapper = shallow(
        <Feat3
          title={title}
          text={text}
          content={mockFeat3Content.content}
          cta={cta}
          images={images}
        />
      );

      const response = wrapper
        .instance()
        .getIconListItems()
        .map(item => {
          return shallow(item).html();
        });

      expect(response).toEqual(expected);
    });
  });
});
