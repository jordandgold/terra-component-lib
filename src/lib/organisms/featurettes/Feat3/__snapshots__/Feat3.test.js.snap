// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Feat3 should match the snapshot with links 1`] = `
<section
  className="ter-feat-three"
>
  <div
    className="ter-feat-three__image-wrapper"
  >
    <img
      alt="placeholder"
      className="ter-feat-three__image ter-feat-three__image--desktop"
      src="./800x1600.png"
    />
    <img
      alt="placeholder"
      className="ter-feat-three__image ter-feat-three__image--tablet"
      src="./800x600.png"
    />
    <img
      alt="placeholder"
      className="ter-feat-three__image ter-feat-three__image--mobile"
      src="./1-to-1.png"
    />
  </div>
  <div
    className="ter-feat-three__content-container"
  >
    <h2
      className="ter-feat-three__title"
    >
      Good design is aesthetic.
    </h2>
    <p
      className="ter-feat-three__text"
    >
      Fugiat ad est sit non elit nulla laborum occaecat proident non. Fugiat non commodo occaecat veniam amet voluptate ea consequat excepteur ea velit. Quis duis fugiat veniam enim velit cupidatat minim. Sint dolor ex eu veniam laborum non elit nulla ea do. Dolor aute eu Lorem reprehenderit adipisicing magna reprehenderit dolore mollit ullamco officia quis ea. Veniam voluptate excepteur aliquip aute aliqua qui cupidatat pariatur. Elit do sint dolore officia duis incididunt duis.
    </p>
    <LinkCard
      key="link-card-0"
      link={
        Object {
          "text": "Button",
          "url": "#",
        }
      }
      text="Cupidatat ea in duis ullamco nostrud cupidatat quis aliqua ex labore occaecat sit. Tempor irure non amet fugiat qui ea occaecat qui. Aliquip esse eu exercitation veniam irure consectetur quis cupidatat incididunt reprehenderit sint."
      title="Good design is innovative"
    />
    <LinkCard
      key="link-card-1"
      link={
        Object {
          "text": "Button",
          "url": "#",
        }
      }
      text="Cupidatat ea in duis ullamco nostrud cupidatat quis aliqua ex labore occaecat sit. Tempor irure non amet fugiat qui ea occaecat qui. Aliquip esse eu exercitation veniam irure consectetur quis cupidatat incididunt reprehenderit sint."
      title="Good design is innovative"
    />
    <LinkCard
      key="link-card-2"
      link={
        Object {
          "text": "Button",
          "url": "#",
        }
      }
      text="Cupidatat ea in duis ullamco nostrud cupidatat quis aliqua ex labore occaecat sit. Tempor irure non amet fugiat qui ea occaecat qui. Aliquip esse eu exercitation veniam irure consectetur quis cupidatat incididunt reprehenderit sint."
      title="Good design is innovative"
    />
    <ButtonLink
      className="ter-button--primary--1"
      link="#"
      text="Sub-CTA"
    />
  </div>
</section>
`;
